### PR TITLE
Don't trigger unecessary konflux builds

### PR DESCRIPTION
**Pull request by sebsoto**
Adds filtering to prevent konflux builds from happening on doc changes

https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds/

### CODE REVIEW

File: .tekton/windows-machine-config-operator-bundle-master-pull-request.yaml [hunk 1]
```diff
@@ -7,8 +7,10 @@ metadata:
     build.appstudio.redhat.com/pull_request_number: '{{pull_request_number}}'
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
-    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
-      == "master"
+    pipelinesascode.tekton.dev/on-cel-expression: |
+      event == "pull_request"
+      && target_branch == "master"
```
Comment (saifshaikh48) [inline .tekton/windows-machine-config-operator-bundle-master-pull-request.yaml:12]: I assume we'll eventually have a konflux config file for each branch? So this value will be release-4.16 once the config for that branch is created, etc?

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm
/hold

holding for @jrvaldes's question (slack)

**Comment by jrvaldes** [comment]
LGTM


@sebsoto  override the e2e tests.

**Comment by sebsoto** [comment]
/override ci/prow/aws-e2e-operator ci/prow/aws-e2e-upgrade ci/prow/azure-e2e-operator ci/prow/azure-e2e-upgrade ci/prow/gcp-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/platform-none-vsphere-e2e-operator ci/prow/vsphere-proxy-e2e-operator ci/prow/vsphere-e2e-operator ci/prow/wicd-unit-vsphere

**Pull request merged**